sql/src/main/java/org/apache/druid/sql/calcite/parser/DruidSqlParserUtils.java

```diff
@@ -38,20 +39,22 @@
 
 public class DruidSqlParserUtils
 {
+
+  private static final Logger log = new Logger(DruidSqlParserUtils.class);
+
   /**
    * Delegates to {@code convertSqlNodeToGranularity} and converts the exceptions to {@link ParseException}
    * with the underlying message
    */
   public static Granularity convertSqlNodeToGranularityThrowingParseExceptions(SqlNode sqlNode) throws ParseException
   {
-    Granularity g;
     try {
-      g = convertSqlNodeToGranularity(sqlNode);
+      return convertSqlNodeToGranularity(sqlNode);
     }
     catch (Exception e) {
+      log.error(e, StringUtils.format("Unable to convert %s to a valid granularity.", sqlNode.toString()));
```

Comment:
If this error occurs, it's due to user error on one particular query, so it shouldn't be an ERROR that appears in the server logs. That should be reserved for really important stuff that cluster operators need to worry about.

However, there should be _some_ way to get the stack trace, and we generally _don't_ want to return it directly to users of the SQL API.

So in a situation like this I'd do `log.debug` instead of `log.error`, so server admins can see the stack trace if they really need to, but won't see it by default. That balances all of the relevant desires: server admins shouldn't see "user error" stuff by default, and users shouldn't see stack traces as part of normal error messages.

Reply:
Updated with the correct level, thanks for the explanation.